packages/flutter_tools/lib/src/resident_devtools_handler.dart

```diff
@@ -81,6 +90,34 @@ class FlutterResidentDevtoolsHandler implements ResidentDevtoolsHandler {
     }
   }
 
+  // This must be guaranteed not to return a Future that fails.
+  @override
+  Future<bool> launchDevToolsInBrowser({@required List<FlutterDevice> flutterDevices}) async {
+    if (!_residentRunner.supportsServiceProtocol || _devToolsLauncher == null) {
+      return false;
+    }
+    if (_devToolsLauncher.devToolsUrl == null) {
+      _logger.startProgress('Waiting for Flutter DevTools to be served...');
+      bool timedOut = false;
+      await _devToolsLauncher.ready.timeout(launchInBrowserTimeout, onTimeout: () {
```

Comment:
15 is arbitrary, but this was to protect against a user hitting the 'v' hotkey before DevTools has finished serving and if for some reason serving were to hang, we wouldn't want to await a never ending future.

Reply:
What behavior would you suggest instead of using a timeout?

Reply:
I suppose you could make it unawaited in general - to let it take as long as it might without blocking the CLI. In the case where the user hits v before serving is finished we should await serving somehow too though.